app/models/user.rb

```diff
@@ -11,10 +11,22 @@ class User < ActiveRecord::Base
 
   has_many :orders, dependent: :destroy
   validates :name, length: { in: 2..32 }
-  validates :username, presence: true, uniqueness: true
+  validates :username, uniqueness: true
   validates :email, presence: true, uniqueness: true,
             format: /\A([^@\s]+)@((?:[-a-z0-9]+\.)+[a-z]{2,})\Z/i
 
   enum role: %w(default admin)
 
+  def self.find_or_create_by_auth(auth_data)
```

Comment:
Assignment Branch Condition size for find_or_create_by_auth is too high. [17.06/15]